core/src/main/java/io/grpc/internal/AutoConfiguredLoadBalancerFactory.java

```diff
@@ -144,48 +148,38 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
       LoadBalancer delegate = getDelegate();
       if (selection.serverList.isEmpty()
           && !delegate.canHandleEmptyAddressListFromNameResolution()) {
-        delegate.handleNameResolutionError(
-            Status.UNAVAILABLE.withDescription(
-                "Name resolver returned no usable address. addrs="
-                + servers + ", attrs=" + attributes));
+        return Status.UNAVAILABLE.withDescription(
+            "NameResolver returned no usable address. addrs=" + servers + ", attrs=" + attributes);
       } else {
         delegate.handleResolvedAddresses(
             ResolvedAddresses.newBuilder()
                 .setAddresses(selection.serverList)
                 .setAttributes(attributes)
                 .build());
+        return Status.OK;
       }
     }
 
-    @Override
-    public void handleNameResolutionError(Status error) {
+    void handleNameResolutionError(Status error) {
       getDelegate().handleNameResolutionError(error);
     }
 
     @Deprecated
-    @Override
-    public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateInfo) {
+    void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateInfo) {
```

Comment:
why cant this be removed yet?

Reply:
LoadBalancers still using the old `createSubchannel()` will still receive subchannel state updates from this path. It will be removed once `LoadBalancer#handleSubchannel` is removed.